.clang-tidy

```diff
@@ -39,9 +50,10 @@ modernize-*,
 performance-*,
 -performance-noexcept-move-constructor,
 -performance-unnecessary-value-param,
+-performance-no-int-to-ptr,
```

Comment:
There were only a couple of instances of this check needing to be nolint? Furthermore, I acutally prefer leaving this check on as you can almost always fix it by replacing an int cast with a ptr_diff or other proper type cast.

Reply:
OK